transport/dlr-malspp/transport/src/main/java/org/ccsds/moims/mo/testbed/util/sppimpl/util/SPPReader.java

```diff
@@ -147,25 +189,30 @@ public synchronized SpacePacket receive() throws IOException
 //        outPacket.setBody(trimmedBody);
     // Read CRC
     if (crcEnabled) {
-      int CRC = SPPHelper.computeCRC(inHeaderBuffer, data, outPacket.getOffset(), dataLength);
-
       is.read(inCrcBuffer);
```

Comment:
Shouldn't be read at the end of the stream if the CRC is not processed.
Also, can we then check if it returns size of inCrcBuffer, and if not, raise an IOException
(could not read CRC, expected 2 bytes, read X)

Reply:
To the if: sure.

to the first comment: I don't understand what you mean. I still need to read the whole packet from the buffer otherwise the later packets get messed up (see above comment).